protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,137 @@
+import {console} from "forge-std/Test.sol";
+import "./IAtomicBridgeInitiator.sol";
+import "./WETH/interfaces/IWETH10.sol";
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+
+contract AtomicBridgeInitiator is IAtomicBridgeInitiator {
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        address recipient;
+        bytes32 hashLock;
+        uint timeLock;
+        bool exists;
+    }
+
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+    mapping(bytes32 => BridgeTransfer) public completedBridgeTransfers;
+    IWETH10 public weth;
+
+    constructor(address _weth) {
+        weth = IWETH10(_weth);
+    }
+
+    function initiateBridgeTransfer(
+        uint256 _wethAmount,
+        address _originator, 
+        address _recipient, 
+        bytes32 _hashLock, 
+        uint _timeLock
+    ) external payable override returns (bytes32 _bridgeTransferId) {
+        console.log("initiateBridgeTransfer");
+        uint256 totalAmount = _wethAmount;
+
+        // If msg.value is greater than 0, convert ETH to WETH and add to total amount
+        if (msg.value > 0) {
+            weth.deposit{value: msg.value}();
+            //Transfer WETH to this contract, revert if transfer fails
+            require(weth.transfer(address(this), msg.value), "WETH transfer failed");
+            totalAmount += msg.value;
+        }
+
+        // Ensure there is a valid total amount
+        require(totalAmount > 0, "Total amount must be greater than 0");
+
+        _bridgeTransferId = keccak256(
+            abi.encodePacked(
+                _originator, 
+                _recipient, 
+                _hashLock, 
+                _timeLock, 
+                block.timestamp
+        ));
+
+        // Check if the bridge transfer already exists
+        require(bridgeTransfers[_bridgeTransferId].amount == 0, "Bridge transfer already exists");
+
+        bridgeTransfers[_bridgeTransferId] = BridgeTransfer({
+            amount: totalAmount,
+            originator: _originator,
```

Comment:
yeah makes sense, if we move `_originator` from the fn sig it will break @cde8's traits for the service. Tagging here so here is aware, remove it from the fn input and `msg.sender`.